### PR TITLE
WIP - Update project setup

### PR DESCRIPTION
**Pull request by pauljamescleary**
This overhaul of the setup takes advantage of several python things.  This could be scaled back, but I found a lot of these useful:

1. `tox` - tox is a great tool for verifying separate python interpreters, everything from python 27, python 36, to pypy and cpython.  I highly recommend using `tox` because we can easily run all of our checks against any python version to ensure compatibility.
1. `setup.cfg` - allows us to configure the "tools" that we are using, like flake8 and pytest
1. `isort` - super cool auto organize your imports.  
1. `coverage` - coverage reporting for source code is also supported.  Easy and simple to post stats to `codecov`.

This specific build uses **appveyor** for most things instead of travis.  The reason is that in order to verify that your stuff works in **Windows**, gotta use appveyor.  I did not configure integration with appveyor as of yet, as we can push off windows support.  However, it is a nice feature, and seemingly very easy to have on.

**Unit tests**

This PR also adds unit tests.  The library is `responses`, and allows you to easily mock requests (and do verification afterwards if you want to).  These are incorporated as pytest fixtures, and make writing unit tests a breeze.  Look at the example unit test under `tests/test_vinyldns.py`.

**What is missing**

1. Should include bumpversion, which auto releases and bumps the version as well with a single command
1. Test signing and releasing to test pypi

**Other notes**

This presently uses `.rst` files instead of markdown `.md` files.  Cannot say I am a huge fan as the syntax is slightly different.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+Changelog
+=========
+
+0.1.0 (2018-11-16)
```
Comment (rebstar6) [inline CHANGELOG.rst:5]: 0.0.1 is up already :P https://pypi.org/project/vinyldns-python/

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,116 @@
+========
+Overview
+========
+
+.. start-badges
```
Comment (rebstar6) [inline README.rst:5]: i made it md intentionally. you can keep with that so long as setup.py is done right

see https://pypi.org/project/markdown-description-example/

File: setup.py [hunk 3]
```diff
@@ -1,42 +1,97 @@
...
+
+def read(*names, **kwargs):
+    with io.open(
+        join(dirname(__file__), *names),
+        encoding=kwargs.get('encoding', 'utf8')
+    ) as fh:
+        return fh.read()
 
-with open('README.md') as f:
-    long_description = f.read()
```
Comment (rebstar6) [inline setup.py:18]: this `with open...` is what let me keep markdown

File: setup.py [hunk 4]
```diff
@@ -1,42 +1,97 @@
...
+    cmdclass={'test': PyTest},
+    name='vinyldns',
+    version='0.1.0',
+    license='Apache Software License 2.0',
+    description='Python client library for VinylDNS',
+    long_description='%s\n%s' % (
+        re.compile('^.. start-badges.*^.. end-badges', re.M | re.S).sub('', read('README.rst')),
+        re.sub(':[a-z]+:`~?(.*?)`', r'``\1``', read('CHANGELOG.rst'))
+    ),
+    author='Paul James Cleary',
```
Comment (rebstar6) [inline setup.py:58]: woah.

File: setup.py [hunk 5]
```diff
@@ -1,42 +1,97 @@
...
+    author_email='pauljamescleary@gmail.com',
```
Comment (britneywright) [inline setup.py:59]: why the author_email change?

**Pull request closed without merging**